core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala

```diff
@@ -154,18 +154,22 @@ abstract class AccumulatorV2[IN, OUT] extends Serializable {
 
   // Called by Java when serializing an object
   final protected def writeReplace(): Any = {
-    if (atDriverSide) {
+    val acc = if (atDriverSide) {
       if (!isRegistered) {
         throw new UnsupportedOperationException(
           "Accumulator must be registered before send to executor")
       }
       val copyAcc = copyAndReset()
       assert(copyAcc.isZero, "copyAndReset must return a zero value copy")
-      copyAcc.metadata = metadata
       copyAcc
     } else {
-      this
+      val copyAcc = copy()
```

Comment:
Do you know it is okay to remove that validation? Just assuming the comments,  it sounds not needed. I just ran `AccumulatorV2Suite` and ` python run-tests.py --module=pyspark-core` and the failures are gone away.